### PR TITLE
New topic docs for internationalisation

### PR DESCRIPTION
**Pull request by kaedroho**
This PR adds new topic-level docs for internationalisation. It covers some explanation about the approach, a how-to guide for initial configuration and some smaller recipe level how to guides.

### CODE REVIEW

File: docs/advanced_topics/i18n.rst [hunk 1]
```diff
@@ -2,22 +2,593 @@
...
+.. code-block:: python
+
+    # my_project/settings.py
+
+    USE_L10N = True
+
+Configuring available languages
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Next we need to configure the available languages, there are two settings
```
Comment (tomdyson) [inline docs/advanced_topics/i18n.rst:158]: languages, there -> languages. There

File: docs/advanced_topics/i18n.rst [hunk 2]
```diff
@@ -2,22 +2,593 @@
...
+Perhaps the most important bit of internationalisation-related UI you can add
+to your site is a selector to allow users to switch between different
+languages.
+
+If you're not convinced that you need this, have a look at https://www.w3.org/International/questions/qa-site-conneg#yyyshortcomings for some rationale.
+
+Basic example
+~~~~~~~~~~~~~
+
+Here is a basic example of how to add linkes between translations of a page.
```
Comment (tomdyson) [inline docs/advanced_topics/i18n.rst:311]: linkes -> links

File: docs/advanced_topics/i18n.rst [hunk 3]
```diff
@@ -2,22 +2,593 @@
...
+.. code-block:: html+Django
+
+    <a href="{% pageurl page.localized %}" rel="alternate" hreflang="{{ language_code }}">
+        {{ lang.name_local }}
+    </a>
+
+The only difference with the ``<a>`` tag here from the ``<a>`` tag in the previous example
+is how we're getting the page's URL: ``{% pageurl page.localized %}``.
+
+All page instnces in Wagtail have a ``.localized`` attribute which fetches the translation
```
Comment (tomdyson) [inline docs/advanced_topics/i18n.rst:438]: instnces -> instances

File: docs/advanced_topics/i18n.rst [hunk 4]
```diff
@@ -2,22 +2,593 @@
...
-Several add-on packages for Wagtail providing multi-language support are available:
+Alternative internationalisation plugins
+========================================
 
-* `Wagtailtrans <https://github.com/wagtail/wagtailtrans>`_
-* `wagtail-modeltranslation <https://github.com/infoportugal/wagtail-modeltranslation>`_
+Before official multi-language support was added into Wagtail, site implementors
+had to use external plugins. These have not been replaced by Wagtail's own
+implementation as they use slightly different approches, so there's a chance that
+onr of them might fit your use case better:
```
Comment (tomdyson) [inline docs/advanced_topics/i18n.rst:585]: onr -> one

File: docs/advanced_topics/i18n.rst [hunk 5]
```diff
@@ -2,22 +2,593 @@
...
 
-Several add-on packages for Wagtail providing multi-language support are available:
+Alternative internationalisation plugins
+========================================
 
-* `Wagtailtrans <https://github.com/wagtail/wagtailtrans>`_
-* `wagtail-modeltranslation <https://github.com/infoportugal/wagtail-modeltranslation>`_
+Before official multi-language support was added into Wagtail, site implementors
+had to use external plugins. These have not been replaced by Wagtail's own
+implementation as they use slightly different approches, so there's a chance that
```
Comment (tomdyson) [inline docs/advanced_topics/i18n.rst:584]: suggest dropping 'there's a chance that'

File: docs/advanced_topics/i18n.rst [hunk 6]
```diff
@@ -2,22 +2,593 @@
...
+        ...
+    {% endfor %}
+
+This ``for`` block iterates through all published translations of the current page.
+
+.. code-block:: html+Django
+
+    {% get_language_info for translation.locale.language_code as lang %}
+
+This is a Django built in tag that gets info about the language of the translation.
```
Comment (tomdyson) [inline docs/advanced_topics/i18n.rst:348]: built in -> built-in

**Review by tomdyson — COMMENTED** [review]
Great job @kaedroho!

File: docs/advanced_topics/i18n.rst [hunk 7]
```diff
@@ -2,22 +2,593 @@
...
+
+Page structure
+^^^^^^^^^^^^^^
+
+Wagtail stores content in a separate page tree for each locale.
+
+For example, if you have two sites in two locales, then you will see four page
+homepages at the top level of the page hierarchy in the explorer.
+
+Translations are linked together in the database so that translations
```
[the remark this answers is not in the document]
Comment (kaedroho) [reply docs/advanced_topics/i18n.rst:49]: I don't either... It's amazing how differently things read sometimes the day after you wrote it!

**Review by gasman — CHANGES_REQUESTED** [review]
Great job! Really clear instructions and explanations here. Just a couple of formatting niggles...

**Comment by kaedroho** [comment]
Thanks @gasman, issues fixed!

**Review by gasman — APPROVED** [review]

**Pull request merged**